### PR TITLE
Add a section to configure password validation criterias

### PR DESCRIPTION
**Pull request by kayathiri4**
### Related PRs
-  (None)

### Checklist
- [ ] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there are any)
- [ ] Unit tests provided. (Add links if there are any)
- [ ] Integration tests provided. (Add links if there are any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [ ] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

**Comment by kayathiri4** [comment]
Pending task: Needs to update the side panel icon : https://github.com/wso2-enterprise/asgardeo-product/issues/13914

File: apps/console/src/features/validation/api/validation-config.ts [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+
+/**
+ * Get an axios instance.
+ *
+ */
+const httpClient = AsgardeoSPAClient.getInstance()
+    .httpRequest.bind(AsgardeoSPAClient.getInstance())
+    .bind(AsgardeoSPAClient.getInstance());
+
+export const updateValidationConfigData = (data: ValidationFormInterface): Promise<ValidationDataInterface[]> => {
```
Comment (brionmario) [inline apps/console/src/features/validation/api/validation-config.ts:40]: Lets add doc comments.

File: apps/console/src/features/validation/pages/validation-config-edit.tsx [hunk 2]
```diff
@@ -0,0 +1,769 @@
...
+                getValidationConfig(rules, "UniqueCharacterValidator", "min.length"): "1",
+            minUpperCaseCharacters: getValidationConfig(rules, "UpperCaseValidator", "min.length") ?
+                getValidationConfig(rules, "UpperCaseValidator", "min.length"): "1",
+            type: "rules",
+            uniqueCharacterValidatorEnabled:
+                getValidationConfig(rules, "UniqueCharacterValidator", "min.unique.character") !== null
+        });
+    };
+
+    const getValidationConfig = (rules: ValidationConfInterface[], validatorName: string,
```
Comment (brionmario) [inline apps/console/src/features/validation/pages/validation-config-edit.tsx:158]: Add doc comments and keep all functions after `useEffect` hooks.

File: apps/console/src/features/validation/pages/validation-config-edit.tsx [hunk 3]
```diff
@@ -0,0 +1,769 @@
...
+                                            id={ FORM_ID }
+                                            initialValues={ initialFormValues }
+                                            uncontrolledForm={ true }
+                                            validate={ null }
+                                            onSubmit={
+                                                (values: ValidationFormInterface) => handleUpdateMyAccountData(values)
+                                            }
+                                            enableReInitialize={ true }
+                                        >
+                                            { /*<Field.Radio*/ }
```
Comment (brionmario) [inline apps/console/src/features/validation/pages/validation-config-edit.tsx:357]: Lets remove the commented code if we are not planning on immediately bring this in. Add a comment in the PR with the commit that has this change.

File: apps/console/src/features/validation/pages/validation-config-edit.tsx [hunk 4]
```diff
@@ -0,0 +1,769 @@
...
+                                            { /*    listen={ (value: string) => {*/ }
+                                            { /*        setRuleType(true);*/ }
+                                            { /*    } }*/ }
+                                            { /*    width={ 16 }*/ }
+                                            { /*    data-testid={ `${ componentId }-rules-toggle` }*/ }
+                                            { /*/>*/ }
+                                            { isRuleType &&
+                                                (<div className="validation-configurations-form">
+                                                    <div className="criteria" >
+                                                        <label>Must be between</label>
```
Comment (brionmario) [inline apps/console/src/features/validation/pages/validation-config-edit.tsx:373]: See if Semantic has a `Form.Label`
```suggestion
                                                        <label for="minLength">Must be between</label>
```

**Review by brionmario — APPROVED** [review]

File: apps/console/src/features/validation/api/validation-config.ts [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+} from "../../../features/core/hooks/use-request";
+import { store } from "../../core";
+import { ValidationManagementConstants } from "../constants/validation-config-constants";
+import { ValidationConfInterface, ValidationDataInterface, ValidationFormInterface } from "../models";
+
+/**
+ * Get an axios instance.
+ *
+ */
+const httpClient = AsgardeoSPAClient.getInstance()
```
Comment (brionmario) [inline apps/console/src/features/validation/api/validation-config.ts:36]:
```suggestion
const httpClient: AsgardeoSPAClient = AsgardeoSPAClient.getInstance()
```

File: apps/console/src/features/validation/pages/validation-config-edit.tsx [hunk 6]
```diff
@@ -0,0 +1,733 @@
...
+    };
+
+    /**
+     * Return values of each validator.
+     *
+     * @param rules - set of configured rules.
+     * @param validatorName - name of the validator.
+     * @param attributeName - name of the attribute.
+     *
+     * @return the value of the configuration.
```
Comment (brionmario) [inline apps/console/src/features/validation/pages/validation-config-edit.tsx:203]:
```suggestion
     * @returns the value of the configuration.
```

File: apps/console/src/features/validation/pages/validation-config.tsx [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+ *
+ * @param props - Props injected to the component.
+ * @returns Governance connector listing page component.
+ */
+export const ValidationConfigPage: FunctionComponent<MyAccountSettingsPageInterface> = (
+    props: MyAccountSettingsPageInterface
+): ReactElement => {
+    const { [ "data-componentid" ]: componentId } = props;
+
+    const pageContextRef = useRef(null);
```
Comment (brionmario) [inline apps/console/src/features/validation/pages/validation-config.tsx:42]:
```suggestion
    const pageContextRef: MutableRefObject<HTMLElement> = useRef(null);
```

**Review by brionmario — APPROVED** [review]

**Review by brionmario — APPROVED** [review]

**Pull request merged**